### ISSUE TITLE
E3009 Additional properties are not allowed ('Fn::Transform' was unexpected)

### ISSUE THREAD

**Issue by rchildress87** [aws-cloudformation/cfn-lint#4363]
### CloudFormation Lint Version

1.44.0

### What operating system are you using?

Alpine (container, from cfn-lint Dockerfile)

### Describe the bug

A template that is using Fn::Transform is giving an error `E3009 Additional properties are not allowed ('Fn::Transform' was unexpected)`. CloudFormation deploys it without complaining. I suspect this has something to do with `Fn::Transform` being under `AWS::CloudFormation::Init` metadata.

### Expected behavior

Template should be parsed without error.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Resources:
  ApplicationInstance:
    Type: AWS::EC2::Instance
    Metadata:
      "AWS::CloudFormation::Init":
        configSets:
          installAndConfigure:
            - installCoreDependencies
            - configureHttpd
        installCoreDependencies:
          packages:
            yum:
              httpd: []
        configureHttpd:
          files:
            /etc/httpd/conf.d/www.conf:
              Fn::Transform:
                Name: AWS::Include
                Parameters:
                  Location: "s3://unique-bucket-name/includes/www.conf.yaml"
              mode: "000644"
              owner: "root"
              group: "root"
    Properties:
      ImageId: ami-0a9f08a6603f3338e
      InstanceType: t4g.micro
      SecurityGroupIds:
        - sg-0b4c66acea364af1
      SubnetId: subnet-0f10a6f6d7cec2aef
      UserData:
        Fn::Base64: !Sub |
          #!/bin/bash -xe

          /opt/aws/bin/cfn-init \
            --stack ${AWS::StackName} \
            --resource ApplicationInstance \
            --configsets installAndConfigure \
            --region ${AWS::Region} \
            --verbose

          /opt/aws/bin/cfn-signal \
            --exit-code $? \
            --stack ${AWS::StackName} \
            --resource ApplicationInstance \
            --region ${AWS::Region}
```

**Comment by rchildress87** [comment]
Hey @kddejong, will you please put eyes on this when you get a chance? I'd like to better understand why this is happening and see how to help.

**Comment by rchildress87** [comment]
Thank you @kddejong !
